dali/pipeline/operator/eager_operator.h

```diff
@@ -165,9 +166,16 @@ class DLL_PUBLIC EagerOperator {
                        ", instance name: \"", name_, "\", encountered:\n", what);
   }
 
-  static inline int GetDefaultNumThreads() {
-    int num_cores = std::thread::hardware_concurrency();
-    return num_cores < 6 ? num_cores : 6;
+  static inline std::shared_ptr<ThreadPool> GetSharedThreadPool() {
+    if (!shared_thread_pool) {
+      int num_cores = std::thread::hardware_concurrency();
+      num_cores = num_cores < 6 ? num_cores : 6;
+
+      shared_thread_pool =
+          std::make_shared<ThreadPool>(num_cores, CPU_ONLY_DEVICE_ID, false, "EagerOperator");
```

Comment:
```suggestion
      UpdateThreadPool(num_cores);
```
?

Also, if it's a member, maybe rename to `shared_thread_pool_`?

Reply:
done